src/components/AddToCart.jsx

```diff
@@ -1,16 +1,30 @@
 import styles from "../styles/AddToCart.module.css";
 
-function addToCart(products, quantity) {
-  const productCount = products.length * quantity;
-  alert(
-    `Added ${productCount} product${
-      productCount !== 1 ? "s" : ""
-    } (${products.join(", ")}) to the cart!`
-  );
+function addToCart(products, quantity, shopify, checkoutId, setCartSize) {
+
+  // TODO create this from the products arg after the storefront id is in the query
+  const lineItemsToAdd = [
+    {
+      variantId: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8zNzExMDI0NDIxMjkwMw==',
+      quantity
+    }
+  ];
+  
+  shopify.checkout.addLineItems(checkoutId, lineItemsToAdd).then((checkout) => {
+    setCartSize(checkout.lineItems.length)
+
+    if (products.length === 1) {
+      alert(`Added ${products[0].title} to the cart!`);
+    } else {
+      alert(
+        `Added ${products.length} products (${products.map(product => product.title).join(", ")}) to the cart!`
+      );
+    }
+  });
 }
 
-const AddToCart = ({ products, label = "Add to cart" }) => {
-  const onClickHandler = () => addToCart(products, 1);
+const AddToCart = ({ products, label = "Add to cart", shopify, checkoutId, setCartSize }) => {
+  const onClickHandler = () => addToCart(products, 1, shopify, checkoutId, setCartSize);
```

Comment:
Instead of passing the cart props all the way down from the `_app` root, I think we could use [React's `useContext` hook](https://reactjs.org/docs/hooks-reference.html#usecontext) to set the value at the root and then pick it up here.

Alternatively, we could also move cart management logic into [Next's API Routes](https://nextjs.org/docs/api-routes/introduction) and then call them using `fetch` requests, which would have the added benefit of keeping our token secret from the client.

Reply:
I'll use context. The `storefrontAccessToken` is not secret and is safe for client code.